test/projectile-test.el

```diff
@@ -87,6 +88,27 @@ You'd normally combine this with `projectile-test-with-sandbox'."
      (spy-on 'projectile-project-root :and-return-value (file-truename (expand-file-name "project/")))
      ,@body))
 
+(defmacro assert-friendly-error-when-no-project (fn)
```

Comment:
To avoid a macro, we could write a single `it` and iterate over a list of functions. However, it would require manually rescuing the error and manually re-throwing the error to include the function name in the error string, to make it clear which function we're dealing with. I wish buttercup would allow us to specify a custom error message, just like rspec does:

```elisp
(expect (funcall 'foo) :to-throw 'error nil "Custom error message here")
```

Reply:
Yeah, that would have been nice. It guess it'd be nice to leave your comment here as comment before the macro, so it's clear why it was needed.